File: simple_linear_regression.py
#Simple Linear Regression

# Importing the libraries

import numpy as np

import matplotlib.pyplot as plt

import pandas as pd


# Importing the dataset

dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:, :-1].values

y = dataset.iloc[:, 1].values


# Splitting the dataset into the Training set and Test set

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2)


# Feature Scaling is done in Simple Regression itself

#Fitting data into Simple regression Model
from sklearn.linear_model import LinearRegression
regressor=LinearRegression()
regressor.fit(X_train,y_train)
#predicting values
y_pred=regressor.predict(X_test)

#visualizing predicted results
plt.scatter(X_train,y_train,color='red')
plt.plot(X_train,regressor.predict(X_train),color="blue")
plt.title("Salary Vs Experience")
plt.xlabel("Experience(in yrs)")
plt.ylabel("Salary")
#visualizing test results
plt.scatter(X_test,y_test,color='red')
plt.plot(X_train,regressor.predict(X_train),color="blue")
plt.title("Salary Vs Experience")
plt.xlabel("Experience(in yrs)")
plt.ylabel("Salary")

File: simple_linear_regression.R
#Simple Linear Regression

#importing dataset
dataset=read.csv('Salary_Data.csv')

#Spliting Data
library(caTools)
split=sample.split(dataset$Salary,SplitRatio=0.8)
training_set=subset(dataset,split==TRUE)
test_set=subset(dataset,split==FALSE)
#Feature Scaling Won't be Needed

#Fitting simple Regression To training set

regressor=lm(formula=Salary~YearsExperience,
             data=training_set)
#Predicting Data 

y_pred=predict(regressor,newdata=test_set)
#Visualizing Training set data
#install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x=training_set$YearsExperience,y=training_set$Salary),
             colour='red')+
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,newdata = training_set)
                ),colour='blue')+
  ggtitle('Salary vs Experience')+
  xlab('Experience')+
  ylab('Salary')
#Visualizing Test Results
ggplot()+
  geom_point(aes(x=test_set$YearsExperience,y=test_set$Salary),colour="red")+
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,newdata = training_set))
            ,colour='blue')+
  ggtitle('Salary vs Experience')+
  xlab('Years Of Experience')+
  ylab('Salary')